code/modules/antagonists/heretic/magic/cosmic_runes.dm

```diff
@@ -0,0 +1,116 @@
+/datum/action/cooldown/spell/cosmic_rune
+	name = "Cosmic Rune"
+	desc = "Create a cosmic rune for teleportation. If there are already two cosmic runes, it destroys the oldest rune."
+	background_icon_state = "bg_heretic"
+	overlay_icon_state = "bg_heretic_border"
+	button_icon = 'icons/mob/actions/actions_ecult.dmi'
+	button_icon_state = "cosmic_rune"
+
+	sound = 'sound/magic/forcewall.ogg'
+	school = SCHOOL_FORBIDDEN
+	cooldown_time = 40 SECONDS
+
+	invocation = "ST'R R'N'"
+	invocation_type = INVOCATION_WHISPER
+	spell_requirements = NONE
+
+	/// Storage for the first rune.
+	var/obj/effect/cosmic_rune/first_rune
+	/// Storage for the second rune.
+	var/obj/effect/cosmic_rune/second_rune
+	/// Rune removal effect.
+	var/obj/effect/rune_remove_effect = /obj/effect/temp_visual/cosmic_rune_fade
+
+/datum/action/cooldown/spell/cosmic_rune/cast(atom/cast_on)
+	. = ..()
+	if(first_rune && second_rune)
+		var/obj/effect/cosmic_rune/new_rune = new /obj/effect/cosmic_rune(get_turf(cast_on))
+		new rune_remove_effect(get_turf(first_rune))
+		QDEL_NULL(first_rune)
+		first_rune = second_rune
+		second_rune = new_rune
+		first_rune.link_rune(null)
+		first_rune.link_rune(second_rune)
+		second_rune.link_rune(first_rune)
+	if(!first_rune)
+		first_rune = new /obj/effect/cosmic_rune(get_turf(cast_on))
+		if(second_rune)
+			first_rune.link_rune(second_rune)
+			second_rune.link_rune(first_rune)
+	else if(!second_rune)
+		second_rune = new /obj/effect/cosmic_rune(get_turf(cast_on))
+		if(first_rune)
+			first_rune.link_rune(second_rune)
+			second_rune.link_rune(first_rune)
+
+/obj/effect/cosmic_rune
+	name = "cosmic rune"
+	desc = "A strange rune, that can instantly transport people to another location."
+	anchored = TRUE
+	icon = 'icons/obj/hand_of_god_structures.dmi'
+	icon_state = "cosmic_rune"
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	layer = SIGIL_LAYER
+	/// The other rune this rune is linked with
+	var/obj/effect/cosmic_rune/linked_rune
+	/// Effect for when someone teleports
+	var/obj/effect/rune_effect = /obj/effect/temp_visual/rune_light
+
+/obj/effect/cosmic_rune/attack_paw(mob/living/user, list/modifiers)
+	return attack_hand(user, modifiers)
+
+/obj/effect/cosmic_rune/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(!linked_rune)
+		to_chat(user, "There must be a second [src] for it to work!")
```

Comment:
```suggestion
		balloon_alert(user, "no linked rune!")
```
For feedback which isn't that important except in the moment, we prefer the use of balloon alerts.